tests/layers/l2_l4_inference_test.py

```diff
@@ -192,6 +204,139 @@ def testCapacity(self):
+      },
+      {
+        0: (locations[3], features[0]),  # SE - round
+        1: (locations[0], features[0]),  # NW - round
+        2: (locations[1], features[1]),  # N  - handle
+        3: (locations[2], features[0]),  # NE - round
+        4: (locations[4], features[0]),  # S  - round
+      },
+      {
+        0: (locations[4], features[0]),  # S  - round
+        1: (locations[3], features[0]),  # SE - round
+        2: (locations[0], features[0]),  # NW - round
+        3: (locations[1], features[1]),  # N  - handle
+        4: (locations[2], features[0]),  # NE - round
+      },
+    ]
+
+    # Box grasp from top
+    # Symetrical features at different locations.
+    objectsToLearn["Box"] = [
+      {
+         # Top/Front of the box
+         0: (locations[5], features[2]),  # W1 - flat
+         1: (locations[6], features[3]),  # N1 - corner
+         2: (locations[7], features[3]),  # N2 - corner
+         3: (locations[8], features[2]),  # E1 - flat
+         4: (locations[9], features[2]),  # E2 - flat
+      },
+      {
+         # Top/Side of the box
+         0: (locations[5], features[2]),   # W1 - flat
+         1: (locations[8], features[2]),   # E1 - flat
+         2: (locations[9], features[2]),   # E2 - flat
+         3: (locations[10], features[2]),  # E3 - flat
+         4: (locations[11], features[2]),  # E4 - flat
+      },
+      {
+         # Top/Back of the box
+         0: (locations[9], features[2]),   # E2 - flat
+         1: (locations[12], features[3]),  # S1 - corner
+         2: (locations[13], features[3]),  # S2 - corner
+         3: (locations[14], features[2]),  # W3 - flat
+         4: (locations[5], features[2]),   # W1 - flat
+      },
+    ]
+
+    # Create 10 experiment instances, train them on all objects and run
+    # inference on one object
+    numExps = 10
+    exps = []
+    for i in range(numExps):
+      exps.append(
+        l2_l4_inference.L4L2Experiment(
+        "testClassification",
+        numCorticalColumns=5,
+        inputSize=1024,
+        numInputBits=20,
+        externalInputSize=1024,
+        numLearningPoints=3,
+        seed=23,
+        )
+      )
+
+      exps[i].learnObjects(objectsToLearn)
+
+      # Try to infer "Mug" using first learned grasp
+      sensations = [
+          objectsToLearn["Mug"][0]
+      ]
+      exps[i].sendReset()
+      exps[i].infer(sensations, reset=False)
+
+
+    # Ensure L2 and L4 representations are consistent across all experiment
+    # instantiations, across all columns, and across 10 different repeats
+    for i in range(10):
+      for c in range(5):
+        total = sum(exps[0].getL2Representations()[c])
+        for e in range(numExps):
+          self.assertEqual(total, sum(exps[e].getL2Representations()[c]))
+
+        total = sum(exps[0].getL4Representations()[c])
+        for e in range(numExps):
+          self.assertEqual(total, sum(exps[e].getL4Representations()[c]))
```

Comment:
Would it make sense to use `set` instead of `sum`? `assertEqual` prints a useful comment when two sets aren't equal. It's harder to make sense of two unequal sums.